java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/utils/ArrowFlightConnectionConfigImpl.java

```diff
@@ -202,9 +202,9 @@ public enum ArrowFlightConnectionProperty implements ConnectionProperty {
     public Object get(final Properties properties) {
       Preconditions.checkNotNull(properties, "Properties cannot be null.");
       Preconditions.checkState(
-          properties.containsKey(camelName) || !required,
+          properties.containsKey(camelName.toLowerCase()) || !required,
```

Comment:
Nit: We should check !required then properties.containsKey()...

Reply:
done